lib/msal-node/src/network/HttpClient.ts

```diff
@@ -267,7 +272,23 @@ const networkRequestViaHttps = <T>(
                     // do not destroy the request for the device code flow
                     networkResponse.body["error"] !== Constants.AUTHORIZATION_PENDING) {
                     request.destroy();
-                    reject(new Error(`HTTP status code ${statusCode}`));
+                    if (networkResponse.body["error"] === InteractionRequiredAuthErrorMessage.invalid_grant.code) {
+                        reject(InteractionRequiredAuthError.createInvalidGrantError(networkResponse.body["error_description"]));
+                    }
+                    else if (networkResponse.body["error"] === InteractionRequiredAuthErrorMessage.noTokensFoundError.code) {
+                        reject(InteractionRequiredAuthError.createNoTokensFoundError());
+                    }
+                    else if (networkResponse.body["error"] === InteractionRequiredAuthErrorMessage.native_account_unavailable.code) {
```

Comment:
These error codes will never come from ESTS. Only `invalid_grant` and `interaction_required` should be transformed into `InteractionRequiredAuthErrorMessage`. I think you have this logic elsewhere in MSAL.